src/state/SceneFooterState.tsx

```diff
@@ -28,6 +38,81 @@ export const [SceneFooterProvider, useSceneFooterState] = createStateProvider(()
   )
 })
 
+export type FooterRender = (sceneWrapperInfo?: SceneWrapperInfo) => React.ReactNode
+const defaultFooterRender: FooterRender = () => null
+
+/**
+ * This is the global provider for the footer render function.
+ */
+export const [SceneFooterRenderProvider, useSceneFooterRenderState] = createStateProvider(() => {
+  const [renderFooter, setRenderFooter] = useState<FooterRender>(() => defaultFooterRender)
+
+  return useMemo(
+    () => ({
+      renderFooter,
+      setRenderFooter
+    }),
+    [renderFooter, setRenderFooter]
+  )
+})
```

Comment:
I'm not sure why we need a separate context here. Could this go in the one above?

I don't think we should fix this right now, though, since it's minor enough and we need to focus on getting the release out.

Reply:
I don't want to re-render all components which use the other context because it may not be an relevant update for them. I also landed on this design when fixing strange infinite loops, so I don't know if moving this state into the other context will cause this infinite loop re-render or not.